demo/src/main/java/com/smashingboxes/surelockdemo/LoginActivity.java

```diff
@@ -106,15 +107,39 @@ protected void onResume() {
                 fingerprintCheckbox.setChecked(false);
             } else {
                 try {
-                    surelock.loginWithFingerprint(KEY_CRE_DEN_TIALS, getFragmentManager(), FINGERPRINT_DIALOG_FRAGMENT_TAG, R.style.SurelockDemoDialog);
+                    setUpBuilder();
+                    builder.loginWithFingerprint(this);
                 } catch (SurelockInvalidKeyException e) {
-                    //TODO show a dialog telling user to re-enter their credentials because their fingerprints have changed.
+                    surelockStorage.clearAll();
+                    showFingerprintLoginInvalidated();
                 }
                 fingerprintCheckbox.setVisibility(View.GONE);
             }
         }
     }
 
+    private void setUpBuilder() {
+        if (builder == null) {
+            builder = new Surelock.SurelockDialogBuilder
+                    (KEY_CRE_DEN_TIALS, getFragmentManager(), FINGERPRINT_DIALOG_FRAGMENT_TAG);
+            builder.withDefaultDialog(R.style.SurelockDemoDialog)
+                    .withKeystoreAlias(KEYSTORE_KEY_ALIAS)
+                    .withSurelockStorage(surelockStorage);
+        }
+    }
```

Comment:
Minor suggestions here:
Could we instead just call this `getSurelockBuilder()` which returns your classes `builder` instance and initializes it if it's null (like you already have)?
And then we can remove the need for a call on a separate line for `setUpBuilder()`. We could instead just call `getSurelockBuilder().loginWithFingerprint(this);` and `getSurelockBuilder().enrollFingerprintAndStore(...)`

Also, one more nit-picky thing: could you rename your variable `builder` to `surelockBuilder`? That way, it's easier for devs to visually see immediately that these are Surelock-related setup steps.